s3/pkg/datastore/yig/utils/global_id.go

```diff
@@ -0,0 +1,74 @@
+package utils
+
+import (
+	"errors"
+	"net"
+	"sync"
+	"time"
+)
+
+const (
+	START_EPOCH     int64 = 1571385784722
+	MACHINE_LEN     uint8 = 10
+	SEQ_LEN         uint8 = 12
+	TIMESTAMP_SHIFS uint8 = 22
+	MAX_MACHINE_NUM int64 = 1023
+	MAX_SEQ_NUM     int64 = 4095
+)
+
+type GlobalIdGen struct {
+	// lower bits of ip address.
+	MachineId int64
+	startTime int64
+	seq       int64
+	mux       sync.Mutex
+}
+
+func NewGlobalIdGen() (*GlobalIdGen, error) {
+	gi := &GlobalIdGen{
+		seq: 0,
+	}
+	id, err := gi.getMachineId()
+	if err != nil {
+		return nil, err
+	}
+	gi.MachineId = id
+	return gi, nil
+}
+
+func (gi *GlobalIdGen) GetId() int64 {
```

Comment:
question...will a running, incrementing sequence number suffice ?  much simpler to handle imho

Reply:
@rhsakarpos because according to snowflake algo, the global unique id has a 'machine id' in it. So, the algo will generate the ids monotonically per machine.